File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/ParticipantStatDao.java
package com.legue.axel.lolsummonertool.database.dao.match;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.match.ParticipantStat;

import java.util.List;

@Dao
public interface ParticipantStatDao {

    @Query("SELECT * FROM participantStats ORDER BY id")
    LiveData<List<ParticipantStat>> getParticipantStats();

    @Query("SELECT * FROM participantStats WHERE participantKey = :participantKey")
    LiveData<ParticipantStat> getParticipantStat(int participantKey);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllParticipantStats(List<ParticipantStat> participantStats);

    @Query("DELETE FROM participantStats")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateParticipantStat(ParticipantStat participantStat);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/ParticipantStatDao.kt
package com.legue.axel.lolsummonertool.database.dao.match

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.match.ParticipantStat

@Dao
interface ParticipantStatDao {

    @Query("SELECT * FROM participantStats ORDER BY id")
    fun getParticipantStats(): LiveData<List<ParticipantStat>>

    @Query("SELECT * FROM participantStats WHERE participantKey = :participantKey")
    fun getParticipantStat(participantKey: Int): LiveData<ParticipantStat>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllParticipantStats(participantStats: List<ParticipantStat>)

    @Query("DELETE FROM participantStats")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateParticipantStat(participantStat: ParticipantStat)
}
